lib/WasmReader/WasmByteCodeGenerator.cpp

```diff
@@ -376,69 +384,96 @@ EmitInfo
-        return EmitConst<WasmTypes::F64>();
+        info = EmitConst<WasmTypes::F64>();
+        break;
     case wnCONST_I32:
-        return EmitConst<WasmTypes::I32>();
+        info = EmitConst<WasmTypes::I32>();
+        break;
     case wnBLOCK:
-        return EmitBlock();
+        info = EmitBlock();
+        break;
     case wnLOOP:
-        return EmitLoop();
+        info = EmitLoop();
+        break;
     case wnCALL_IMPORT:
-        return EmitCall<wnCALL_IMPORT>();
+        info = EmitCall<wnCALL_IMPORT>();
+        break;
     case wnCALL:
-        return EmitCall<wnCALL>();
+        info = EmitCall<wnCALL>();
+        break;
     case wnCALL_INDIRECT:
-        return EmitCall<wnCALL_INDIRECT>();
+        info = EmitCall<wnCALL_INDIRECT>();
+        break;
     case wnIF:
-        return EmitIfExpr();
-    case wnIF_ELSE:
-        return EmitIfElseExpr();
+        info = EmitIfElseExpr();
+        break;
+    case wnELSE:
+        throw WasmCompilationException(_u("Unexpected else opcode"));
+    case wnEND:
+        throw WasmCompilationException(_u("Unexpected end opcode"));
     case wnBR:
-        return EmitBr<wnBR>();
+        info = EmitBr<wnBR>();
+        break;
     case wnBR_IF:
-        return EmitBr<wnBR_IF>();
+        info = EmitBr<wnBR_IF>();
+        break;
     case wnSELECT:
-        return EmitSelect();
+        info = EmitSelect();
+        break;
     case wnBR_TABLE:
-        return EmitBrTable();
+        info = EmitBrTable();
+        break;
     case wnNOP:
-        return EmitInfo();
+        info = EmitInfo();
+        break;
 #define WASM_MEMREAD(token, name, type) \
     case wn##token: \
-        return EmitMemRead<wn##token, WasmTypes::##type>();
+        info = EmitMemRead<wn##token, WasmTypes::##type>(); \
+        break;
 #define WASM_MEMSTORE(token, name, type) \
     case wn##token: \
-        return EmitMemStore<wn##token, WasmTypes::##type>();
+        info = EmitMemStore<wn##token, WasmTypes::##type>(); \
+        break;
 #define WASM_KEYWORD_BIN_TYPED(token, name, op, resultType, lhsType, rhsType) \
     case wn##token: \
-        return EmitBinExpr<Js::OpCodeAsmJs::##op, WasmTypes::##resultType, WasmTypes::##lhsType, WasmTypes::##rhsType>();
-
+        info = EmitBinExpr<Js::OpCodeAsmJs::##op, WasmTypes::##resultType, WasmTypes::##lhsType, WasmTypes::##rhsType>(); \
+        break;
 #define WASM_KEYWORD_UNARY(token, name, op, resultType, inputType) \
     case wn##token: \
-        return EmitUnaryExpr<Js::OpCodeAsmJs::##op, WasmTypes::##resultType, WasmTypes::##inputType>();
-
+        info = EmitUnaryExpr<Js::OpCodeAsmJs::##op, WasmTypes::##resultType, WasmTypes::##inputType>(); \
+        break;
 #include "WasmKeywords.h"
-
+    case wnNYI:
+        // todo:: add the name of the operator that is NYI
+        throw WasmCompilationException(_u("Operator NYI"));
     default:
         throw WasmCompilationException(_u("Unknown expression's op 0x%X"), op);
     }
+    PushEvalStack(info);
```

Comment:
i am wondering if the expressiveness we allow is strictly greater than what the spec allows? it seems we would allow something like:
set_local a 1
set_local b 1
call hello
add